controllers/helpers.go

```diff
@@ -253,6 +253,8 @@ func newCloudProviderConfig(d azure.ClusterScoper) (controlPlaneConfig *CloudPro
 			SecurityGroupName:            subnet.SecurityGroup.Name,
 			SecurityGroupResourceGroup:   d.Vnet().ResourceGroup,
 			Location:                     d.Location(),
+			ExtendedLocationType:         d.ExtendedLocationType(),
```

Comment:
is this supported in all versions of cloud provider both in tree and out of tree? If not what's the minimum version that supports it?

Reply:
It is not supported in "in-tree" cloud provider, but supported in "out-of-tree" cloud provider since version v1.0.3.